var/spack/repos/builtin/packages/chapel/package.py

```diff
@@ -3,24 +3,385 @@
+        "1.28.0",
+        deprecated=True,
+        sha256="321243a91f8f2dfb3b37a714e2d45298e6a967a9a115565f9ad9cc630ff0bd0e",
+    )
+    # Do NOT add older versions here.
+    # Chapel releases over 2 years old are not supported.
+
+    depends_on("doxygen@1.8.17:")
+
+    variant(
+        "llvm",
+        default="unset",
+        description="LLVM backend type. Use value 'spack' to have spack "
+        "handle the LLVM package",
+        values=("none", "system", "bundled", "spack", "unset"),
+    )
+
+    variant(
+        "comm",
+        default="none",
+        description="Build Chapel with multi-locale support",
+        values=("none", "gasnet", "ofi"),
+    )
+    variant(
+        "substrate",
+        default="none",
+        description="Build Chapel with mulit-locale support using the "
+        "supplied CHPL_COMM_SUBSTRATE",
+        values=("none", "udp", "ibv", "ofi"),
+        multi=False,
+    )
+
+    package_module_opts = ("zmq", "libevent", "protobuf", "ssl", "hdf5", "yaml")
+    package_module_dict = {
+        "zmq": "libzmq",
+        "libevent": "libevent",
+        "protobuf": "protobuf",
+        "ssl": "openssl",
+        "hdf5": "hdf5+hl~mpi",
+        "yaml": "libyaml",
+    }
+
+    variant(
+        "package_modules",
+        description="Include package module dependencies with spack",
+        values=disjoint_sets(("none",), ("all",), package_module_opts)
+        .with_error("'none' or 'all' cannot be activated along with other package_modules")
+        .with_default("none")
+        .with_non_feature_values("none", "all"),
+    )
+
+    for opt, dep in package_module_dict.items():
+        depends_on(dep, when="package_modules={0}".format(opt), type="run")
+        depends_on(dep, when="package_modules=all", type="run")
+
+    platform_opts = (
+        "unset",
+        "cygwin32",
+        "cygwin64",
+        "darwin",
+        "linux32",
+        "linux64",
+        "netbsd32",
+        "netbsd64",
+        "pwr6",
+        "cray-cs",
+        "cray-xc",
+        "hpe-apollo",
+        "hpe-cray-ex",
+    )
+
+    variant(
+        "host_platform",
+        description="Host platform",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
+
+    variant(
+        "target_platform",
+        description="Target platform for cross compilation",
+        default="unset",
+        values=platform_opts,
+        multi=False,
+    )
+
+    variant(
+        "tasks",
+        description="Select tasking layer for intra-locale parallelism",
+        default="qthreads",
+        values=("qthreads", "fifo"),
+        multi=False,
+    )
+
+    variant(
+        "re2",
+        description="Build with re2 support",
+        default="bundled",
+        values=("none", "bundled"),
+        multi=False,
+    )
+
+    variant(
+        "gmp",
+        description="Build with gmp support",
+        default="unset",
+        values=("unset", "system", "none", "bundled"),
```

Comment:
Would using a Spack GMP be a reasonable default for this one?

Reply:
I think spack would be happiest if we just used their packages over our bundled software in most cases. It looks like I can add a spack option here.

Reply:
My initial attempts to add spack GMP is giving me problems, not finding `gmp.h` unless I add it to `CPATH`, then complaining about not finding `libgmp` even though I have added its `lib/` to `LD_LIBRARY_PATH` and confirmed `libgmp.so` and `.a` exist there. I'm still working on it, just noting it here along with another thing - the GMP spack package is "currently looking for a maintainer" and is back at 6.2.1 (though I don't necessarily think that's the problem or an incompatibility, just a note)

Reply:
In that case I would think that leaving it unset is reasonable for now. IIRC, that will use a system GMP if we find one that is suitable, and if not, use the bundled GMP.